pyccel/parser/base.py

```diff
@@ -190,18 +190,44 @@ def scope(self, scope):
 
     @property
     def filename(self):
+        """
+        The file being translated.
+
+        Get the name of the file being translated.
+        """
         return self._filename
 
     @property
     def code(self):
+        """
+        The original code.
+
+        Get the original Python code which is being translated.
+        """
         return self._code
 
     @property
     def fst(self):
+        """
+        Full syntax tree.
+
+        Get the full syntax tree describing the code. This object contains `PyccelAstNode`s
+        and is generated by the semantic stage. The full syntax tree is similar to the
+        abstract syntax tree, but additionally contains information about the types of the
+        objects etc.
+        """
         return self._fst
 
     @property
     def ast(self):
+        """
+        Abstract syntax tree.
+
+        Get the abstract syntax tree describing the code. This object contains `PyccelAstNode`s
+        and is generated by the syntactic stage. The abstract syntax tree is similar to the
+        full syntax tree, but only contains infomation about the syntax, there is no semantic
+        data (e.g. the types of variables are unknown).
+        """
```

Comment:
I see a contradiction here. The PR description reads:
> [...] and use ast when talking about objects defined by Python's ast library.

Reply:
The PR description is talking about `pyccel.ast.basic.PyccelAstNode.ast`. This is the output of the SyntacticParser. I agree that it would be better to give this a different name for clarity. Do you have any suggestions?

Reply:
Do you mind if I do this the other way round and use `pyccel_ast` for a `PyccelAstNode`? This matches the names more closely. Plus `.ast` is used a lot more frequently for Python's AST

Reply:
If we use `pyccel_ast` then if would be natural to have a `pyccel_fst` as well. But that sounds a bit overkill.

Reply:
I've had a look. It seems that we are not clear in our use of `ast` and `fst`. So I have changed `ast`->`python_ast`. I will unravel the ast/fst mess in another PR

Reply:
[2f3f9ff](https://github.com/pyccel/pyccel/pull/1593/commits/2f3f9ff395be737f6f48702b296090bea082ba92)